apps/ledger-live-desktop/src/internal/transportHandler.ts

```diff
@@ -0,0 +1,108 @@
+import { Subject } from "rxjs";
+import { withDevice } from "@ledgerhq/live-common/hw/deviceAccess";
+
+type APDUMessage = { apduHex: string; requestId: string };
+const transports = new Map<string, Subject<APDUMessage>>();
+
+export const transportOpenChannel = "transport:open";
+export const transportExchangeChannel = "transport:exchange";
+export const transportCloseChannel = "transport:close";
+
+export const transportOpen = ({
+  data,
+  requestId,
+}: {
+  data: { descriptor: string };
+  requestId: string;
+}) => {
+  const subjectExist = transports.get(data.descriptor);
+  if (subjectExist) {
+    // Should we return saying the transport is open ?
+    process.send?.({
+      type: transportOpenChannel,
+      error: new Error("Transport already opened for the given descriptor"),
+      requestId,
+    });
+    return;
+  }
+
+  withDevice(data.descriptor)(transport => {
+    const subject = new Subject<APDUMessage>();
+    transports.set(data.descriptor, subject);
+
+    subject.subscribe({
+      next: e => {
+        transport
+          .exchange(Buffer.from(e.apduHex, "hex"))
+          .then(response =>
+            process.send?.({
+              type: transportExchangeChannel,
+              data: response.toString("hex"),
+              requestId: e.requestId,
+            }),
+          )
+          .catch(error =>
+            process.send?.({ type: transportExchangeChannel, error, requestId: e.requestId }),
```

Comment:
you may want to serializeError to mirror the deserializeError (until we find a better way, but at least this way was proven working and was done in the "commands")